### ISSUE TITLE
Upgrade the plugin API for stop

### ISSUE THREAD

**Issue by vincent-zurczak** [roboconf/roboconf-platform#320]
Currently, when a mandatory dependency is removed, the agent stops the dependent instance and sets the **UNRESOLVED** state. However, it is not possible for the plugin to determine whether it is stopped normally or because one of its dependencies was removed.

With Petals, we face the case where the stop/start might be heavy in terms of performance (because it means registering all the services on every start). It would be very appreciated to determine whether the stop is normal or due to the disparition of a dependency. This information should be passed to the agent as a flag. Just like we did for the **update** operation, we could pass the removed import as a parameter.

It is then up to the plug-ins (script, Puppet) to use this information (or not).
So, basically, this issue is about...

* ... replacing **stop( Instance i )** by **stop( Instance i, Import removed )**.
* ... passing a new env variable to the stop script in the **script** plug-in.
* ... passing a new parameter to the Puppet hash in the **puppet** plug-in.
